cluster/charts/rook-ceph-cluster/templates/_helpers.tpl

```diff
@@ -4,3 +4,41 @@ Define the clusterName as defaulting to the release namespace
 {{- define "clusterName" -}}
 {{ .Values.clusterName | default .Release.Namespace }}
 {{- end -}}
+
+{{/*
+Return the target Kubernetes version
+*/}}
+{{- define "capabilities.kubeVersion" -}}
+{{- if .Values.global }}
+    {{- if .Values.global.kubeVersion }}
+    {{- .Values.global.kubeVersion -}}
+    {{- else }}
+    {{- default .Capabilities.KubeVersion.Version .Values.kubeVersion -}}
+    {{- end -}}
+{{- else }}
+{{- default .Capabilities.KubeVersion.Version .Values.kubeVersion -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the appropriate apiVersion for ingress.
+*/}}
+{{- define "capabilities.ingress.apiVersion" -}}
+{{- if .Values.ingress -}}
+{{- if .Values.ingress.apiVersion -}}
+{{- .Values.ingress.apiVersion -}}
+{{- else if semverCompare "<1.14-0" (include "capabilities.kubeVersion" .) -}}
+{{- print "extensions/v1beta1" -}}
```

Comment:
Our minimum k8s version is v15, correct @travisn ?

If so, we don't need the condition for `<1.14`.

Reply:
minimum documented version is 1.11 according to [this documentation](https://rook.io/docs/rook/v1.7/pre-reqs.html#minimum-version)

Reply:
Correct, 1.11 is still the min version for Rook v1.7. The min version is bumped up to K8s 1.16 in Rook v1.8.

Reply:
I can simplify this section more by removing the .Values.ingress.apiVersion as well if you think it is suitable

Reply:
I guess this might be backported, so we may as well leave it.

Reply:
I've removed it anyway for simplicity. I dont think this pr should be back ported as the changes here are only relevant for 1.7 because that is when this helm chart was introduced. The bug this solves was introduced in 1.7.3 so it should be fine to go into 1.7.8